chalicelib/es_connection.py

```diff
@@ -0,0 +1,279 @@
+
+    ESConnection is intended to work with only a single index.
+
+    Implements the AbstractConnection 'interface'
+    """
+    def __init__(self, index=None, doc_type='result'):
+        self.es = es_utils.create_es_client(HOST, use_aws_url=True)
+        self.index = index
+        self.mapping = {} # only needed if we create the index
+        if index and not self.index_exists(index):
+            self.create_index(index)
+        self.doc_type = doc_type
+
+    def index_exists(self, name):
+        """
+        Checks if the given index name exists
+        """
+        return self.es.indices.exists(index=name)
+
+    def create_index(self, name):
+        """
+        Creates an ES index called name. Returns true in success
+        """
+        try:
+            self.mapping = self.load_mapping()
+            self.es.indices.create(index=name,body={
+                "settings": {
+                    "index.mapper.dynamic": False
+                },
+                "mappings": self.mapping.get('mappings')
+            }, ignore=400)
+            return True
+        except Exception as e:
+            raise ElasticsearchException(str(e))
+
+    def load_mapping(self, fname='mapping.json'):
+        """
+        Loads ES mapping from 'mapping.json' or another relative path from this
+        file location.
+        """
+        return load_json(__file__, fname)
+
+    def delete_index(self, name):
+        """
+        Deletes the given index name from this es
+        """
+        try:
+            self.es.indices.delete(index=name, ignore=[400, 404])
+        except:
+            return False
+        return True
+
+    def refresh_index(self):
+        """
+        Refreshes the index, then waits 3 seconds
+        """
+        self.es.indices.refresh(index=self.index)
+        time.sleep(3)
+
+    def put_object(self, key, value):
+        """
+        Index a new item into es. Returns true in success
+        """
+        if not self.index:
+            return False
+        try:
+            res = self.es.index(index=self.index, id=key, doc_type=self.doc_type, body=value)
+            return res['result'] == 'created'
+        except Exception as e:
+            print('Failed to add object id: %s with error: %s' % (key, str(e)))
+            return False
+
+    def get_object(self, key):
+        """
+        Gets object with uuid=key from es. Returns None if not found or no index
+        has been specified.
+        """
+        if not self.index:
+            return None
+        try:
+            return self.es.get(index=self.index, doc_type=self.doc_type, id=key)['_source']
+        except:
+            return None
+
+    def get_size(self):
+        """
+        Returns the number of items indexed on this es instance. Returns -1 in
+        failure.
+        """
+        try:
+            return self.es.count(self.index).get('count')
+        except:
+            return -1
```

Comment:
Maybe 0 would make more sense here? It is falsey whereas -1 is not (e.g. `-1 != False`)